cypress/page/personal-form.page.ts

```diff
@@ -27,71 +29,33 @@ class PersonalFormPage {
+    public visitPersonalInfoFormPage(): void{
+        cy.visit(this.personalInfoFormUrl);
+    }
+
+    public fillForm(personalInformation: PersonalFormPage.PersonalInformation): void {
         cy.get(this.nameInput).type(personalInformation.name);
         cy.get(this.lastnameInput).type(personalInformation.lastName);
         cy.get(this.emailInput).type(personalInformation.email);
         cy.get(this.genderRadioBtn).check(personalInformation.gender, {force: true});
         this.fillDateInput(personalInformation.dateOfBirth);
         cy.get(this.mobileNumberInput).type(personalInformation.mobileNumber);
-        personalInformation.hobbies.forEach(hobbie =>
-            {
-                cy.get(this.hobbiesCheckbox).filter(`:contains("${hobbie}")`).click();
-            });
+        personalInformation.hobbies.forEach(hobbie =>{
+            cy.get(this.hobbiesCheckbox).filter(`:contains("${hobbie}")`).click();
+        });
         cy.get(this.currentAddressInput).type(personalInformation.currentAddress);
         cy.get(this.submitBtn).click({force: true});
     }
 
     public fillDateInput(date: string){
-        let year;
-        let month;
-        let day;
+        const monthNames = ["January", "February", "March", "April", "May", "June",
+        "July", "August", "September", "October", "November", "December"
+        ];
+        const newdate = new Date(date);
+        const year = newdate.getFullYear();
+        const month = monthNames[newdate.getMonth()];
+        const day = newdate.getDate();
         cy.get(this.dateOfBirthPicker).click();
-        if (date.length == 10) {
-            day = date[0]
-            month = date.substring(2, 5)
-            year = date.substring(6)
-        }
-        else {
-            day = date.substring(0, 1)
-            month = date.substring(3, 6)
-            year = date.substring(7)
-        }
-        switch (month) {
-            case "Jan":
-                month = "January";
-                break;
-            case "Feb":
-                month = "February";
-                break;
-            case "Mar":
-                month = "March";
-                break;
-            case "Apr":
-                month = "April";
-                break;
-            case "Jun":
-                month = "June";
-                break;
-            case "Jul":
-                month = "July";
-                break;
-            case "Aug":
-                month = "August";
-                break;
-            case "Sep":
-                month = "September";
-                break;
-            case "Oct":
-                month = "October";
-                break;
-            case "Nov":
-                month = "November";
-                break;
-            case "Dic":
-                month = "Dicember";
-                break;
-        }
         cy.get(this.datePickerValues[0]).select(`${year}`);
```

Comment:
can you convert the `datePickerValues` into an object so instead of referring to indexes we can use dot notation: `this.datePickerValues.year`